[docs] Add EAS Tutorial (#27059)
expo/expo: docs/pages/tutorial/eas/using-github.mdx

---
title: Trigger builds from a GitHub repository
sidebar_title: Builds from GitHub
description: Learn about the process of triggering builds from a GitHub repository.
---

import { Step } from '~/ui/components/Step';
import ImageSpotlight from '~/components/plugins/ImageSpotlight';
import { BoxLink } from '~/ui/components/BoxLink';
import { BookOpen02Icon } from '@expo/styleguide-icons';
import { ProgressTracker } from '~/ui/components/ProgressTracker';

[Expo GitHub App](/build/building-from-github/) automatically triggers builds from our GitHub projects with EAS. We can trigger builds for any build profile based on our development team's preferences. It also allows triggering builds for `git` push committed directly to a repository or a pull request.

> In this chapter, we're to configure this functionality. We already have a GitHub repository for our example app to demonstrate this.

Author: ```suggestion
In this chapter, we'll configure this functionality. We already have a GitHub repository for our example app to demonstrate this.
```

The Expo GitHub app needs to know where to find the source code of our project. By default, it selects the root directory using `/`. In our example project, the source code is also available in the root repository.

<ImageSpotlight
  alt="Default settings for configuring a GitHub repository in Expo dashboard"
  src="/static/images/tutorial/eas/github-03.png"
  style={{ maxWidth: 800 }}
/>

</Step>

<Step label="4">

## Trigger a build using a GitHub PR label

The Expo GitHub app provides us [multiple options](/build/building-from-github/#trigger-a-build-from-github) to trigger a build, such as:

- Manually from the Builds page for a specific platform
- Automatically when new code is pushed to the repository
- Automatically using GitHub PR labels

To automatically trigger a build using a GitHub PR label, we're going to utilize the third option from the list above:

- We need to specify the build image that we will be using. Open **eas.json**, and under the `development` profile, add [`android.image`](/eas/json/#image) and [`ios.image`](/eas/json/#image-1) properties and set their value to [`latest`](/build-reference/infrastructure/#configuring-build-environment).

> ```json eas.json
> {
>   "build": {
>     "development": {
>       /* @hide ... */ /* @end */
>       /* @info */
>       "android": {
>         "image": "latest"
>       },
>       "ios": {
>         "image": "latest"
>       }
>       /* @end */
>     }
>   }
>   /* @hide ... */ /* @end */
> }
> ```

Reviewer: ```suggestion
  ```json eas.json
  {
    "build": {
      "development": {
        /* @hide ... */ /* @end */
        /* @info */
        "android": {
          "image": "latest"
        },
        "ios": {
          "image": "latest"
        }
        /* @end */
      }
    }
    /* @hide ... */ /* @end */
  }
  ```
```

Formatting nit, this should indent code block to match list bullet.

> - Next, modify our app's JavaScript code. We create a git branch called `dev` and submit a PR to our example repository from that branch.

Reviewer 2: ```suggestion
- Next, create a new branch called `dev`, make a change in your app's JavaScript code, commit it, push the branch and create a PR from that branch.
```

Author: Thanks for the suggestion! I slightly modified it to keep the second person voice intact here (hope you won't mind).
